music21/stream/__init__.py

```diff
@@ -1668,8 +1666,19 @@ def setElementOffset(
         setActiveSite=True
     ):
         '''
+        DEPRECATED!
+        MOVED in v6.7 to :meth:`~music21.stream.core.coreSetElementOffset`
+        This backwards-compatible shim may be removed in the future.
+
         Sets the Offset for an element, very quickly.
+        Caller is responsible for calling :meth:`~music21.stream.core.coreElementsChanged`
+        afterward.
 
+        See docs on :meth:`~music21.stream.core.coreSetElementOffset`
```

Comment:
Question that I really should know the answer -- is there any other non-core way to do something like this on a stream:

```
for i, el in enumerate([n1, n2, n3, n4]):
     s.setElementOffset(el, i)
```
I know we can do:

```
for i, el in enumerate([n1, n2, n3, n4]):
     el.setOffsetBySite(s, i)
```
but that is an expensive operation.

So, I don't think we should deprecate setElementOffset, only deprecate the addElement keyword (definitely) and probably setActiveSite keyword as well.  Then this method would simply be:

```
    def setElementOffset(
        self,
        element: base.Music21Object,
        offset: Union[int, float, Fraction, str],
        *,
        addElement=False,  # deprecated
        setActiveSite=True  # deprecated
     ):
     
         self.coreSetElementOffset(element,
                                   offset,
                                   addElement=addElement,  # to become always False in v7
                                   setActiveSite=setActiveSite)  # to become always True in v7
         self.coreElementsChanged(updateIsFlat=False)  # might change sorting, but not flatness.  Maybe other things can be False too.
```

When deprecating an entire function, there is a common.decorators.deprecated decorator that I put on the function saying when it will be deprecated and disappear.  It handles all the warnings, etc.  But just store that away because we won't deprecate.